### PR TITLE
Benchmarking :  Fix server response aggregation

### PR DESCRIPTION
**Pull request by varun-sundar-rabindranath**
Summary:
  In Streaming mode, the vllm server returns responses as soon as a token is available. However, it doesn't do it in parts, instead, each response is already an aggregate of all the previous responses. Therefore, it is sufficient to record just the last response.

Test:
Manual testing

### CODE REVIEW

File: neuralmagic/benchmarks/scripts/backend_request_func.py [hunk 1]
```diff
@@ -71,12 +77,12 @@ async def async_request_tgi(
...
             async with session.post(url=api_url, json=payload) as response:
                 if response.status == 200:
-                    data = ""
+                    data = None
                     async for part_data in response.content.iter_any():
                         if ttft == 0:
                             ttft = time.perf_counter() - st
                             output.ttft = ttft
-                        data += part_data
+                        data = part_data
```
Comment (mgoin) [inline neuralmagic/benchmarks/scripts/backend_request_func.py:85]: this doesn't not seem right as you will overwrite with part_data each time

**Review by andy-neuma — APPROVED** [review]
cool.

File: neuralmagic/benchmarks/scripts/backend_request_func.py (hunk 1 above)
Comment (varun-sundar-rabindranath) [reply neuralmagic/benchmarks/scripts/backend_request_func.py:85]: yeah - the thing is, part_data always has all the previous outputs - I verified it by printing out `part_data`. 

Also note that this script is copied from vllm upstream and getting the last `part_data` was the original functionality (https://github.com/vllm-project/vllm/blob/57f044945f25d90d1b434014b2719ba6b06fdc44/benchmarks/backend_request_func.py#L61)
I added the `data += part_data` and introduced a bug - this is basically a reversion.

File: neuralmagic/benchmarks/scripts/backend_request_func.py [hunk 2]
```diff
@@ -43,6 +43,12 @@ def trim_prefix(text: str, prefix_str: str) -> str:
     return text[len(prefix_str):]
 
 
+def trim_suffix(text: str, suffix_str: str) -> str:
+    assert len(text) >= len(suffix_str)
+    assert text[-1 * len(suffix_str):] == suffix_str
+    return text[:-1 * len(suffix_str)]
```
Comment (mgoin) [inline neuralmagic/benchmarks/scripts/backend_request_func.py:49]: nit: why does this require that the suffix exists and fail if it doesn't? i would simplify it to this if the assertion isn't required
```suggestion
def trim_suffix(text: str, suffix_str: str) -> str:
    if text.endswith(suffix_str):
        return text[:-len(suffix_str)]
    return text
```

**Review by mgoin — APPROVED** [review]

File: neuralmagic/benchmarks/scripts/backend_request_func.py (hunk 2 above)
Comment (varun-sundar-rabindranath) [reply neuralmagic/benchmarks/scripts/backend_request_func.py:49]: It is out of the timing code and I think is good to keep around so we know we aren't trimming any actual data.

**Pull request merged**